cmd/docker/docker.go

```diff
@@ -313,6 +324,7 @@ func areFlagsSupported(cmd *cobra.Command, details versionDetails) error {
 			if _, ok := f.Annotations["experimentalCLI"]; ok && !hasExperimentalCLI {
 				errs = append(errs, fmt.Sprintf("\"--%s\" is on a Docker cli with experimental cli features enabled", f.Name))
 			}
+			// buildkit-specific flags are noop when buildkit is not enabled, so we do not add an error in that case
```

Comment:
Maybe we can still print `logrus.Warn` warnings, especially for  NOP `--squash` with BuildKit mode

Reply:
Could make sense to inform them that an option is only supported with/without buildkit, and will be ignored.

Given that we're already ignoring those flags in the current code, I think that'd be good to do separately, in a follow-up;

```
DOCKER_BUILDKIT=1 docker build --squash .
[+] Building 0.3s (4/4) FINISHED                                                                                                                                                                      
 => local://context (.dockerignore)                                                                                                                                                              0.0s
 => => transferring context: 2B                                                                                                                                                                  0.0s
 => local://dockerfile (Dockerfile)                                                                                                                                                              0.0s
 => => transferring dockerfile: 31B                                                                                                                                                              0.0s
 => CACHED docker-image://docker.io/library/alpine:latest                                                                                                                                        0.0s
 => exporting to image                                                                                                                                                                           0.0s
 => => exporting layers                                                                                                                                                                          0.0s
 => => writing image sha256:0b4ed36840824a86314da6264a95047c664f3c2fe90d0b1a6ddb118e20f106a2                  
```

```
DOCKER_BUILDKIT=0 docker build --progress=plain .
Sending build context to Docker daemon  2.048kB
Step 1/1 : FROM alpine
 ---> 11cd0b38bc3c
Successfully built 11cd0b38bc3c
```